src/cartpole/dircol_swingup.py

```diff
@@ -44,13 +50,28 @@
 result = Solve(dircol)
 assert result.is_success()
 
-x_trajectory = dircol.ReconstructStateTrajectory(result)
-
 fig, ax = plt.subplots(2, 1)
 
-vis = PlanarRigidBodyVisualizer(tree, xlim=[-1.5, 1.5], ylim=[-.5, 2],
-                                ax=ax[0])
-ani = vis.animate(x_trajectory, repeat=True)
+# Animate the results.
+# TODO(russt): Add some helper methods to make this workflow cleaner.
+def animate(plant, x_trajectory):
```

Comment:
FYI -- @SeanCurtis-TRI.  Just an example of what we were talking about.

I actually tried creating the `SceneGraph` and call `plant.RegisterAsSourceForSceneGraph` down here, just to see.  The code ran but the animation was black (presumably no geometry actually registered).  But the RegisterAsSourceForSceneGraph method documentation did not actually say that it had to be called pre-finalize... eventually I found that the MultibodyPlant class docs do.  Perhaps we should add a note?

Reply:
or even throw an error?

Reply:
I'm not entirely sure what it is I'm supposed to infer from this code example. As I read the documentation, `RegisterAsSourceForSceneGraph()` [pretty obviously requires that it be called pre-finalized](https://github.com/RobotLocomotion/drake/blob/master/multibody/plant/multibody_plant.h#L971).

There are numerous aspects of MBP's relationship with SG that I intend/hope to change. They are hacks and stopgaps that have sadly hardened somewhat with time. I'm looking to start whittling away at them. One of the biggest ones is that MBP knows way to much about geometry. But that's not your point.

In my intended/proposed decoupling, you *should* be able to do this *after* the fact without difficulties (i.e., your plant is your plant, and later on you can decide, "Hey, it'd be nice to do something with `SceneGraph`. I'll do that now." The primary challenge behind that is if the MBP contents arose from parsing multiple files. I'm still reasoning through that part.

Reply:
You're right.  Sorry I missed the post-finalize note.
What to infer -- just that this is my workflow right now.  It's a bit clunky, but it works.